### DISCUSSION TITLE
Incompatibilty Error from Base Classes with `**kwargs`

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [python/typing#1235] [Q&A]
Related to [this SO question](https://stackoverflow.com/questions/53017505/mypy-method-definition-is-incompatible-with-definition-in-base-class), I'm trying to understand why

- [this doesn't yield errors](https://mypy-play.net/?mypy=latest&python=3.8&gist=103fe953e90ce3b1f7bd09ef3a3c6297)

but

- [adding `**kwargs` to the end of the `run` functions does](https://mypy-play.net/?mypy=latest&python=3.8&gist=ca1267a10d1746d0677e23306df98013)

**Comment by adam-grant-hendry** [comment]
I believe, if I'm not mistaken, the error could be due to the potential for a keyword collision, which I gleaned from [PEP 692](https://peps.python.org/pep-0692/#keyword-collisions). Confirmation from the community would be appreciated though!

e.g. If I made [the name of the 4th arguments the same](https://mypy-play.net/?mypy=latest&python=3.8&gist=2cba6dbcf026d3b6b769a53c9e42ce76), each could fail the same way if I tried to pass `foo` as a keyword argument:

```python
>>> a = Foo()
>>> a.run(1, 2, 'baz', foo='bar')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: func() got multiple values for argument 'foo'
```

but if I keep the argument names different, then the above would work if I tried to call `Bar.run(1, 2, 'baz', foo='bar')`, but break for `Foo`.

**Comment by JelleZijlstra** [comment]
I don't think there's a very principled answer here: strictly speaking both should be an error. I believe pyright will flag both as errors but I haven't tried.

Mypy is fairly lenient with the names of positional parameters in overrides, in part because mypy's design predates convenient syntax for positional-only parameters. That's why your first example doesn't give an error.

**Reply by adam-grant-hendry** [reply]
@JelleZijlstra Thank you for your reply.

I can copy-paste the examples to the `pyright` playground and see what they say.

This is a source of confusion for me and others ([see this SO post](https://stackoverflow.com/questions/53017505/mypy-method-definition-is-incompatible-with-definition-in-base-class)). If you would be able to weigh in, it would be much appreciated!

**Reply by adam-grant-hendry** [reply]
@JelleZijlstra On the [pyright playground](https://pyright-playground.decorator-factory.su/?issue=microsoft/pyright/2662), the OP code has no errors, which is what I would expect since nothing is statically typed in that code so everything should be read as type `Any`.

Is this by chance a `mypy` bug? Should I file an issue from this discussion?
